### ISSUE TITLE
Merged PR 223 breaks previously working rust build

### ISSUE THREAD

**Issue by johnae** [nix-community/dream2nix#251]
When building https://github.com/nushell/nushell I get failing builds after #223 

I.e, this fails:

```nix
{
  description = "Test";

  inputs = {
    ## flakes
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    flake-utils.url = "github:numtide/flake-utils";
    dream2nix = {
      url = "github:nix-community/dream2nix/600f213c7cc76b43c3e41abf5215cbf37940a38c";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    nushell = {
      url = "github:nushell/nushell/0.66.2";
      flake = false;
    };
  };

  outputs = {
    self,
    nixpkgs,
    flake-utils,
    dream2nix,
    ...
  } @ inputs: let
    l = nixpkgs.lib // builtins;

    d2n = rec {
      init = pkgs:
        dream2nix.lib.init {
          inherit pkgs;
          config.projectRoot = ./.;
        };
      makeOutputs = {
        pkgs,
        source,
        packageOverrides ? {},
      }: let
        outputs = (init pkgs).makeOutputs {
          inherit source packageOverrides;
        };
      in {
        inherit (outputs) packages;
      };
    };

    d2nPackages = flake-utils.lib.eachDefaultSystem (
      system: let
        pkgs = nixpkgs.legacyPackages.${system};
      in
        d2n.makeOutputs {
          inherit pkgs;
          source = inputs.nushell;
          packageOverrides."^.*".addDeps = {
            nativeBuildInputs = old: old ++ [pkgs.pkgconfig pkgs.python3];
            buildInputs = old: old ++ [pkgs.openssl pkgs.zstd pkgs.xorg.libX11];
            doCheck = false;
            buildFeatures = ["extra"];
            cargoUpdateHook = ''
              cargo add zstd-sys --features pkg-config --offline
              cargo update --package zstd-sys --offline
            '';
          };
        }
    );
  in
    d2nPackages;
}

```

with:

```sh
❯ nix build '.#nu'
error: builder for '/nix/store/x1jvyx0sn5zcfcsyqnf1g3n1x282axq5-nu-0.66.2.drv' failed with exit code 101;
       last 10 log lines:
       > unpacking source archive /nix/store/1gn9q6n274r8h4p18g9238qyfh4ar3ak-source/
       > source root is source
       > Executing cargoSetupPostUnpackHook
       > Finished cargoSetupPostUnpackHook
       > patching sources
       > configuring
       > building
       > Executing cargoBuildHook
       > ++ env CC_x86_64-unknown-linux-gnu=/nix/store/kia8xygy2r9iblwbjrl163j4bpz84wz7-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/store/kia8xygy2r9iblwbjrl
163j4bpz84wz7-gcc-wrapper-11.3.0/bin/c++ CC_x86_64-unknown-linux-gnu=/nix/store/kia8xygy2r9iblwbjrl163j4bpz84wz7-gcc-wrapper-11.3.0/bin/cc CXX_x86_64-unknown-linux-gnu=/nix/
store/kia8xygy2r9iblwbjrl163j4bpz84wz7-gcc-wrapper-11.3.0/bin/c++ cargo build -j 32 --target x86_64-unknown-linux-gnu --frozen --release --package nu
       > error: package collision in the lockfile: packages nu-cli v0.66.2 (/build/source/crates/nu-cli) and nu-cli v0.66.2 (/nix/store/1gn9q6n274r8h4p18g9238qyfh4ar3ak-sour
ce/crates/nu-cli) are different, but only one can be written to lockfile unambiguously
       For full logs, run 'nix log /nix/store/x1jvyx0sn5zcfcsyqnf1g3n1x282axq5-nu-0.66.2.drv'.
```

while this succeeds (eg. the only change is referencing the commit before merging #223):

```nix
{
  description = "Test";

  inputs = {
    ## flakes
    nixpkgs.url = "github:nixos/nixpkgs/nixos-unstable";
    flake-utils.url = "github:numtide/flake-utils";
    dream2nix = {
      url = "github:nix-community/dream2nix/4a92c17f9579fa49667cb6cba8ad63acdfb1eb32";
      inputs.nixpkgs.follows = "nixpkgs";
    };
    nushell = {
      url = "github:nushell/nushell/0.66.2";
      flake = false;
    };
  };

  outputs = {
    self,
    nixpkgs,
    flake-utils,
    dream2nix,
    ...
  } @ inputs: let
    l = nixpkgs.lib // builtins;

    d2n = rec {
      init = pkgs:
        dream2nix.lib.init {
          inherit pkgs;
          config.projectRoot = ./.;
        };
      makeOutputs = {
        pkgs,
        source,
        packageOverrides ? {},
      }: let
        outputs = (init pkgs).makeOutputs {
          inherit source packageOverrides;
        };
      in {
        inherit (outputs) packages;
      };
    };

    d2nPackages = flake-utils.lib.eachDefaultSystem (
      system: let
        pkgs = nixpkgs.legacyPackages.${system};
      in
        d2n.makeOutputs {
          inherit pkgs;
          source = inputs.nushell;
          packageOverrides."^.*".addDeps = {
            nativeBuildInputs = old: old ++ [pkgs.pkgconfig pkgs.python3];
            buildInputs = old: old ++ [pkgs.openssl pkgs.zstd pkgs.xorg.libX11];
            doCheck = false;
            buildFeatures = ["extra"];
            cargoUpdateHook = ''
              cargo add zstd-sys --features pkg-config --offline
              cargo update --package zstd-sys --offline
            '';
          };
        }
    );
  in
    d2nPackages;
}
```

**Comment by yusdacra** [comment]
Does this fix it for you? https://github.com/nix-community/dream2nix/pull/252

**Comment by yusdacra** [comment]
Fixed via #252
